File: src/main/java/org/jcodec/containers/mp4/boxes/URIMetaSampleEntry.java
package org.jcodec.containers.mp4.boxes;

import java.nio.ByteBuffer;

public class URIMetaSampleEntry extends MetaDataSampleEntry {
    public URIMetaSampleEntry(Header atom) {
        super(atom);
    }

    public void parse(ByteBuffer input) {
        super.parse(input);
        parseExtensions(input);
    }

    protected void doWrite(ByteBuffer out) {
        super.doWrite(out);
        writeExtensions(out);
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/URIMetaSampleEntry.kt
package org.jcodec.containers.mp4.boxes

import java.nio.ByteBuffer

class URIMetaSampleEntry(atom: Header) : MetaDataSampleEntry(atom) {
    override fun parse(input: ByteBuffer) {
        super.parse(input)
        parseExtensions(input)
    }

    override fun doWrite(out: ByteBuffer) {
        super.doWrite(out)
        writeExtensions(out)
    }
}